### PR TITLE
[REVIEW] Hjiang/pr prune parquet bloom filter date

### PR DESCRIPTION
**Pull request by dentiny**
> [!NOTE]
> **Medium Risk**
> Incorrect unit/hash mapping could falsely exclude row groups and drop matching rows; the optional-hash fallback limits that, but temporal encoding edge cases need the new test coverage to hold.
> 
> **Overview**
> Extends Parquet **bloom-filter row-group pruning** beyond numeric/string types by making bloom support and hashing **schema-aware** via `ParquetColumnSchema` instead of `LogicalTypeId` alone.
> 
> `BloomFilterSupported` / `BloomFilterExcludes` (and `parquet_bloom_probe`) now take the column schema. Filter constants are hashed with `ValueXXH64(constant, schema)`, which maps DuckDB values to the **encoded Parquet representation** (e.g. millisecond timestamps divided from microseconds, time units, finite vs infinite timestamp sentinels). Unsupported encodings return no hash so pruning is skipped rather than guessed wrong.
> 
> `BloomFilterSupported` adds **DATE** and temporal types gated on Parquet physical type and unit (ms/micros/ns); some TIME/TIME_TZ ns encodings stay unsupported when sub-microsecond precision cannot be matched.
> 
> Tests cover temporal bloom probes, **EXPLAIN ANALYZE** expecting zero row groups scanned for missing values, TIMETZ UTC normalization, and infinity timestamp filters on ms Parquet files.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit e911a76e8a39e88ba0b7129b421c639e33e891d8. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: extension/parquet/parquet_statistics.cpp [hunk 1]
```diff
@@ -728,10 +729,94 @@ static uint64_t ValueXH64FixedWidth(const Value &constant) {
...
+}
+
+// Recreate the Parquet millisecond hash input from DuckDB's microsecond timestamp representation.
+static optional<uint64_t> TryHashTimestamp(const Value &constant, const ParquetColumnSchema &schema) {
+	D_ASSERT(constant.type().id() == LogicalTypeId::TIMESTAMP || constant.type().id() == LogicalTypeId::TIMESTAMP_TZ);
+	auto value = constant.GetValue<int64_t>();
+	switch (schema.type_info) {
+	case ParquetExtraTypeInfo::UNIT_MS:
+		// DuckDB stores timestamps in microseconds, while the Parquet bloom filter hashes the stored milliseconds.
+		if (timestamp_t {value}.IsFinite()) {
```
Comment (dentiny) [inline extension/parquet/parquet_statistics.cpp:793]: Need to understand
